src/main/java/in/ac/bits/protocolanalyzer/persistence/repository/AnalysisRepository.java

```diff
@@ -73,7 +73,7 @@ public void save(IndexQuery query) {
 	public void start() {
 		log.info("Starting analysis repository...");
 		TimerTask pull = new TimerTask() {
-
+			
 			@Override
 			public void run() {
 				while (!queries.isEmpty()) {
```

Comment:
Line No 85, why aren't we using queries.take() instead of queries.poll()?

Reply:
Since we are checking for non-zero size of the ArrayBlockingQueue, both poll() and take() have the same effect.

Reply:
So the issue faced with this function is that, if I remove Thread.delay code from its corresponding Test function ("saveAnalysisTest") the corresponding asserts on the ArrayBlockingQueue queries and  ArrayList currentBucket fail and only occasionally pass. Only upon providing the delay the correct functioning is carried out by "public void start".